game-core/src/main/java/games/strategy/triplea/attachments/TechAbilityAttachment.java

```diff
@@ -570,17 +465,7 @@ public int getWarBondDiceNumber() {
   }
 
   public static int getWarBondDiceNumber(final PlayerID player, final GameData data) {
-    int warBondDiceNumber = 0;
-    for (final TechAdvance ta : TechTracker.getCurrentTechAdvances(player, data)) {
-      final TechAbilityAttachment taa = TechAbilityAttachment.get(ta);
-      if (taa != null) {
-        final int number = taa.getWarBondDiceNumber();
-        if (number > 0) {
-          warBondDiceNumber += number;
-        }
-      }
-    }
-    return Math.max(0, warBondDiceNumber);
+    return sumNumbers(TechAbilityAttachment::getWarBondDiceNumber, player, data);
```

Comment:
I noticed that `sumNumbers()` doesn't contain the `Math.max(0, ...)` condition.  Is it not necessary simply because you are ensuring all values are positive before summing?  Do we need to consider the case where the sum overflows, and thus will wrap back to a negative value?

Reply:
>  Is it not necessary simply because you are ensuring all values are positive before summing?

Exactly

> Do we need to consider the case where the sum overflows, and thus will wrap back to a negative value?

I really don't think so 😄, besides that I really don't think a 0 if too high makes much sense either